clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -226,11 +246,20 @@ public void commitAsync(OffsetCommitCallback callback) {
 
     @Override
     public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCommitCallback callback) {
-        final CommitApplicationEvent commitEvent = new CommitApplicationEvent(offsets);
-        commitEvent.future().whenComplete((r, t) -> {
-            callback.onComplete(offsets, new RuntimeException(t));
+        CompletableFuture<Void> future = commit(offsets);
+        future.whenComplete((r, t) -> {
+            if (t != null) {
+                callback.onComplete(offsets, new RuntimeException(t));
+            } else {
+                callback.onComplete(offsets, null);
+            }
         });
+    }
+
+    private CompletableFuture<Void> commit(Map<TopicPartition, OffsetAndMetadata> offsets) {
```

Comment:
I have a KIP proposal here: 
1. Can we introduce a commit interface to return a CompletableFuture?
2. Do we want to introduce a poll interface returning a CompletableFuture? (Like do we really need it to block?)

Reply:
I thought we have those methods on the existing Consumer interface. Anyway, that doesn't impact this PR.